### PR TITLE
feat(incognia.go): adds new header with the user-agent information

### PR DESCRIPTION
**Pull request by joaoaab**
This PR adds a new header containing a 'User-Agent' header, it should contain the following format:

\<library-name\>/\<library-version\> (\<os-name\> \<os-version\> \<os-arch\>) \<language-name\>/\<language-version\>
i got the following outcome running locally 
"incognia-go/1.8.0 (GNU/Linux 5.13.0-39-generic x86_64) golang/go1.17"

### CODE REVIEW

**Review by figueredo — APPROVED** [review]

File: incognia.go [hunk 1]
```diff
@@ -7,11 +7,16 @@ import (
...
+	"runtime"
+	"syscall"
 	"time"
 )
 
 const (
 	defaultNetClientTimeout = 5 * time.Second
+	contentTypeKey          = "Content-Type"
+	userAgentKey            = "User-Agent"
+	libVersion              = "1.8.0"
```
Comment (t0rr3sp3dr0) [inline incognia.go:19]: This shouldn't be hardcoded and must be injected in compile time. Something like:

```bash
go build -ldflags="-X 'repo.incognia.com/go/incognia.libVersion=$(git describe --always --dirty --long)'"
```

Example:
https://github.com/inloco/entrypoint-tool/blob/615aee22aabb6f7126f6224745296c6c6a95326d/Makefile#L26-L36
https://github.com/inloco/entrypoint-tool/blob/615aee22aabb6f7126f6224745296c6c6a95326d/version/main.go#L8

File: incognia.go [hunk 2]
```diff
@@ -437,3 +445,32 @@ func isValidFeedbackType(feedbackType FeedbackType) bool {
...
+	headers[userAgentKey] = generateUserAgentString()
+	return headers
+}
+
+func generateUserAgentString() string {
+	osName := runtime.GOOS
+	osArch := runtime.GOARCH
+
+	var uts syscall.Utsname
+	syscall.Uname(&uts)
```
Comment (t0rr3sp3dr0) [inline incognia.go:461]: The syscall package is OS specific and the uname syscall is Linux specific. By adding this here you are breaking compatibility with macOS and Windows.

You should either use a OS independent function to get this information or implement this also for macOS and Windows, falling back to an empty string on other OSes.

**Review by t0rr3sp3dr0 — CHANGES_REQUESTED** [review]

File: incognia.go (hunk 2 above)
Comment (joaoaab) [reply incognia.go:461]: hmm, great call

File: incognia.go (hunk 1 above)
Comment (joaoaab) [reply incognia.go:19]: Okay, i'm on it

File: incognia.go [hunk 3]
```diff
@@ -7,11 +7,17 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"runtime"
 	"time"
+
+	"github.com/matishsiao/goInfo"
```
Comment (t0rr3sp3dr0) [inline incognia.go:13]: I just took a look at this package and it doesn't comply with our standards.

Actually, this is so simple that we can write these functions ourselves. I don't believe we need a dependency for this.

**Review by t0rr3sp3dr0 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**